src/Microsoft.VisualStudio.Threading/AwaitExtensions.cs

```diff
@@ -49,6 +49,17 @@ public static TaskSchedulerAwaitable SwitchTo(this TaskScheduler scheduler, bool
             return new TaskSchedulerAwaitable(scheduler, alwaysYield);
         }
 
+        /// <summary>
+        /// Gets an awaitable that schedules continuations on the specified synchronization context.
+        /// </summary>
+        /// <param name="context">The synchronization context used to execute continuations.</param>
+        /// <returns>An awaitable.</returns>
+        public static SynchronizationContextAwaiter GetAwaiter(this SynchronizationContext context)
```

Comment:
As discussed yesterday, let's keep this internal. Folks who await a SynchronizationContext (or a TaskScheduler) may deadlock on shutdown because they're not leveraging JTF. So it's not a pattern we want to encourage.

---
In reply to: [200798277](https://github.com/Microsoft/vs-threading/pull/313#discussion_r200798277) [](ancestors = 200798277)